src/torrent.cpp

```diff
@@ -4040,22 +4040,45 @@ namespace {
 				ret = false;
 				break;
 			}
-			auto result = get_hash_picker().set_block_hash(piece, i * default_block_size, block_hashes[i]);
+			auto const result = get_hash_picker().set_block_hash(piece
+				, i * default_block_size, block_hashes[i]);
 
-			block_passed[i] = result.status == set_block_hash_result::result::success;
```

Comment:
when we allocate `block_passed`, we clear it to all be `false`. Setting it to false here should not be necessary. What's left is setting it to `true`, which we can do unconditionally in the if-block below. But, in fact, we already do a `fill_n()` to set fields to `true` inside the if-block, and the range we set should always include `i`, which means we don't actually need this assignment anymore.

Reply:
Right. IIRC the verified block range and corresponding fill_n came much later than this line. I clearly didn't notice that the former obsoleted the latter.